### PR TITLE
Implement simple moving average

### LINKED ISSUE #6112: Support simple moving average aggregate function
This issue is for implementing the simple moving average aggregate function.

Syntax:

```
# Take a moving average from field using 2 values
SELECT moving_average(value, 2) FROM cpu
# Take the median every 10 seconds and get a moving average from the last minute every 10 seconds
SELECT moving_average(median(value), 6) FROM cpu GROUP BY time(10s)
```

Related to #77.

### PR DESCRIPTION
**Pull request by jsternberg**
Fixes #6112.

### CODE REVIEW

**Comment by jsternberg** [comment]
Requires #6105 to be merged before this one is merged.

**Comment by jsternberg** [comment]
@nathanielc

File: influxql/functions.go [hunk 1]
```diff
@@ -53,3 +53,75 @@ func (r *IntegerMeanReducer) Emit() []FloatPoint {
...
+		r.points = append(r.points, FloatPoint{
+			Value:      r.sum / float64(len(r.buf)),
+			Time:       p.Time,
+			Aggregated: uint32(len(r.buf)),
+		})
+	}
+}
+
+func (r *FloatMovingAverageReducer) Emit() []FloatPoint {
+	return r.points
```
Comment (nathanielc) [inline influxql/functions.go]: Why are we returning a set of points? Wouldn't we just return a single point with the current average?

Ah I think I get it, the Aggregate method is called exactly `n` times so we only emit one point?

Comment (nathanielc) [inline influxql/functions.go]: I get it now, the points list grows but becomes an down-sampled version of the original data set. The end result is N/n where N is the original data set size and n is the window size.

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,7 @@
 - [#6060](https://github.com/influxdata/influxdb/pull/6060): Add configurable shard duration to retention policies
 - [#6116](https://github.com/influxdata/influxdb/pull/6116): Allow `httpd` service to be extensible for routes
 - [#6111](https://github.com/influxdata/influxdb/pull/6111): Add ability to build static assest. Improved handling of TAR and ZIP package outputs.
```
Comment (e-dard) [inline CHANGELOG.md]: Do you not want to reference you're adding SMA here?

**Comment by e-dard** [comment]
LGTM :+1:

File: CHANGELOG.md (hunk 2 above)
Comment (jsternberg) [inline CHANGELOG.md]: Yep, I'll add that after the difference PR gets merged (since this one is dependent on it).

**Comment by jsternberg** [comment]
@e-dard updated the PR with a changelog and an integration test.

I didn't add an integration test for fill though. Do you think it's necessary for me to add that?

**Comment by jsternberg** [comment]
I added the fill tests too just to be complete.

**Comment by jwilder** [comment]
:+1:

**Pull request merged**